feat(astro): address astro env rfc feedback (#11213)
withastro/astro: .changeset/dirty-rabbits-act.md

---
'astro': patch
---

**BREAKING CHANGE to the experimental `astro:env` feature only**

Server secrets specified in the schema must now be imported from `astro:env/server`. Using `getSecret()` is no longer required to use these environment variables in your schema:

```diff
- import { getSecret } from 'astro:env/server'
- const API_SECRET = getSecret("API_SECRET")
+ import { API_SECRET } from 'astro:env/server'
```

> Note that using `getSecret()` with these keys no longer involves any special handling and the raw value will be returned, just like retrieving secrets not specified in your schema.

Reviewer: ```suggestion
Note that using `getSecret()` with these keys is still possible, but no longer involves any special handling and the raw value will be returned, just like retrieving secrets not specified in your schema.
```

---
'astro': patch
---

**BREAKING CHANGE to the experimental `astro:env` feature only**

> Server secrets specfied in the schema must now be imported from `astro:env/server` along with `getSecret()`:

Author: ```suggestion
Server secrets specified in the schema must now be imported from `astro:env/server`. Using `getSecret()` with these keys no longer involves any special handling and the raw value will be returned.

Note that `getSecret()` can still be used to retrieve secrets not specified in your schema.
```

```diff
- import { getSecret } from 'astro:env/server'

> + import { FOO, getSecret } from 'astro:env/server'
> 
> - const FOO = getSecret("FOO")

Reviewer: ```suggestion
+ import { API_SECRET, getSecret } from 'astro:env/server'

- const API_SECRET = getSecret("API_SECRET")
``` 
We always prefer to use somewhat meaningful names!

const UNKNOWN = getSecret("UNKNOWN")
```

> 
> Note that `getSecret` can still be used to retrieve secrets not specified in your schema. If the key is also in the schema, no special handling will occur and the raw value will be returned.

Author: ```suggestion
```
